pkg/auth/artifact_verifier.go

```diff
@@ -115,50 +126,60 @@ func NewBuildManifestVerifier(keyringPath string, fetcher uri.Fetcher, logger *l
 
 // Returns an error if the stanza's artifact is not signed appropriately. Note that this
 // implementation does not use the pod manifest digest location options.
-func (b *BuildManifestVerifier) VerifyHoistArtifact(localCopy *os.File, artifactLocation *url.URL) error {
-	switch artifactLocation.Scheme {
-	default:
-		return util.Errorf("%v does not have a recognized scheme '%v', cannot verify manifest or signature", artifactLocation.String(), artifactLocation.Scheme)
-	case "http", "https", "file":
-		dir, err := ioutil.TempDir("", "artifact_verification")
-		if err != nil {
-			return util.Errorf("Could not create temporary directory for manifest file: %v", err)
-		}
-		defer os.RemoveAll(dir)
+func (b *BuildManifestVerifier) VerifyHoistArtifact(localCopy *os.File, verificationData VerificationData) error {
+	manifestLocation := verificationData.ManifestLocation
+	if manifestLocation == nil {
+		return util.Errorf("Manifest verification failed: manifest location not provided")
+	}
 
-		manifestSrc := &url.URL{}
-		*manifestSrc = *artifactLocation
-		manifestSrc.Path = manifestSrc.Path + ".manifest"
+	manifestSignatureLocation := verificationData.ManifestSignatureLocation
+	if manifestSignatureLocation == nil {
+		return util.Errorf("Manifest verification failed: manifest signature location not provided")
+	}
 
-		manifestDst := filepath.Join(dir, "manifest")
+	if manifestLocation.Scheme != "http" &&
+		manifestLocation.Scheme != "https" &&
+		manifestLocation.Scheme != "file" {
+		return util.Errorf("%v does not have a recognized scheme '%v', cannot verify manifest or signature", manifestLocation.String(), manifestLocation.Scheme)
```

Comment:
I realize that it was present before in the `switch` statement, but now that we've decided on the `version` key, instead of a `image://` protocol, I don't think it's necessary